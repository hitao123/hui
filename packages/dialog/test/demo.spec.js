import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Dialog from '../dialog.vue';
import DialogDemo from '../demo/index.vue';

describe('Dialog component', () => {
  it('should show Dialog', () => {
    const warpper = shallowMount(Dialog, {
      propsData: {
        title: 'title',
        cancelButtonText: 'cancel',
        confirmButtonText: 'confirm'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show Button Demo', () => {
    expect(renderToString(DialogDemo)).toMatchSnapshot();
  });
});
