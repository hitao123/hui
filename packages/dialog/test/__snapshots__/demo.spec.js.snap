// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component should show Button Demo 1`] = `
<div class="demo-dialog">
  <h-button>点击dialog</h-button>
  <h-button>点击dialog2</h-button>
  <h-dialog show-cancel-button="" message="&lt;div style=&quot;font-weight: bold;&quot;&gt;message&lt;/div&gt;" title="title" cancelButtonText="cancel" confirmButtonText="confirm"></h-dialog>
</div>
`;

exports[`Dialog component should show Dialog 1`] = `
<div class="h-dialog" style="display: none;" name="h-dialog-bounce">
  <div class="h-dialog__header">title</div>
  <!---->
  <div class="h-hairline--top h-dialog__footer">
    <h-button-stub tag="button" type="default" size="large" class="h-dialog__cancel" style="display: none;">
      cancel
    </h-button-stub>
    <h-button-stub tag="button" type="default" size="large" class="h-dialog__confirm">
      confirm
    </h-button-stub>
  </div>
</div>
`;
